File: net/cm-api-v1/PutItem.cs
// DocSection: cm_api_v1_put_item
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Management;

ManagementOptions options = new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
};

ManagementClient client = new ManagementClient(options);

string externalId = "59713";

ContentItemUpsertModel item = new ContentItemUpsertModel()
{
    Name = "On Roasts",
    Type = ContentTypeIdentifier.ByCodename("article"),
  	SitemapLocations = new[] { SitemapNodeIdentifier.ByCodename("articles") }
};

ContentItemModel contentItemResponse = await client.UpsertContentItemByExternalIdAsync(externalId, item);
// EndDocSection

File: js/management-api-v2/PutItem.js
// DocSection: cm_api_v2_put_item
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

// Updates an existing content item
const response = await client.updateContentItem()
  .byItemId('f4b3fc05-e988-4dae-9ac1-a94aba566474')
  // .byItemCodename('my_article')
  // .byItemExternalId('59713')
  .withData(
    {
      name: 'On Roasts',
      codename: 'my_article_my_article'
    }
  )
  .toPromise();

// Creates a new content item
const response = await client.upsertContentItem()
  .byItemExternalId('59713')
  .withData(
    {
      name: 'On Roasts',
      codename: 'my_article_my_article',
      // 'type' is only required when creating a new content item
      type: 'article'
    }
  )
  .toPromise();
// EndDocSection